### PR TITLE
UCSC-166 Scaffolding and Featured block

### PR DESCRIPTION
**Pull request by MlKilderkin**
## What does this do/fix?

Register Query_Loop middleware class 
Add Featured Block 

### Links to relevant issues

- https://moderntribe.atlassian.net/browse/UCSC-166
- https://moderntribe.atlassian.net/browse/UCSC-172

### CODE REVIEW

File: src/Blocks/Query_Loop.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+	/**
+	 * Default label for single card on Manual Query
+	 * Add/Change in Block clack in order to override
+	 */
+	protected string $default_manual_card_label = 'Card';
+
+	/**
+	 * Default max amount of cards applied to the manual query
+	 * Add/Change in Block clack in order to override
```
Comment (bjcooper) [inline src/Blocks/Query_Loop.php:31]: What is "clack"?

Comment (bjcooper) [reply src/Blocks/Query_Loop.php:31]: Is it supposed to be "class"? Also in the subsequent block comments.

File: src/Core.php [hunk 2]
```diff
@@ -3,36 +3,80 @@
...
 	protected function blocks(): void {
-		add_action( 'init', static function (): void {
-			register_block_type( UCSC_DIR . '/src/views/news_block' );
-			( new News_Block() )->init();
+		add_action( 'init', function (): void {
+			$this->init_blocks();
 			( new News_Blocks_Hooks() )->hooks();
 			( new Assets_Subscriber() )->register();
+
+			if ( ! defined( 'UCSC_NEWS_SITE' ) && ! UCSC_NEWS_SITE ) {
```
Comment (bjcooper) [inline src/Core.php:32]: Doesn't this check if the `UCSC_NEWS_SITE` constant is not defined and then, if it's not, check it's value?

```suggestion
			if ( ! defined( 'UCSC_NEWS_SITE' ) || ! UCSC_NEWS_SITE ) {
```

File: src/Core.php [hunk 3]
```diff
@@ -3,36 +3,80 @@
...
 				'ucsc-news-block-scripts',
-                UCSC_PLUGIN_URL . '/assets/js/news-block.js',
+				UCSC_PLUGIN_URL . '/assets/js/news-block.js',
 				[],
 				false,
 				true
 			);
 			wp_enqueue_script( 'ucsc-news-block-scripts' );
+
+			if ( ! defined( 'UCSC_NEWS_SITE' ) && ! UCSC_NEWS_SITE ) {
```
Comment (bjcooper) [inline src/Core.php:51]: Same as above. Please ignore me if I'm just not understanding what this does.

File: src/Core.php [hunk 4]
```diff
@@ -3,36 +3,80 @@
...
 		}, 10, 0 );
 	}
+	
+	protected function init_blocks(): void {
+		foreach ( self::BLOCKS_PUBLIC as $block_class => $block_path ) {
+			register_block_type( UCSC_DIR . $block_path );
+			( new $block_class )->init();
+		}
+		
+		if ( ! defined( 'UCSC_NEWS_SITE' ) && ! UCSC_NEWS_SITE ) {
```
Comment (bjcooper) [inline src/Core.php:72]: Same as above.

**Review by Vinsanity — APPROVED** [review]
Just a few small things to clean up and answer here Myk and we should be ok.

File: src/Blocks/Query_Loop.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+			],
+			'default_value' => self::AUTOMATIC,
+		];
+	}
+	
+	protected function get_automatic_query(): array {
+		return [
+			'key'               => $this->get_field_key( self::AUTOMATIC_GROUP, $this->block_name ),
+			'type'              => 'group',
+			'label'             => esc_html__( '', 'ucsc' ),
```
[the remark this answers is not in the document]
Comment (MlKilderkin) [reply src/Blocks/Query_Loop.php:88]: Yes. If we add label here it will look a bit odd on interface

File: src/Core.php (hunk 2 above)
Comment (MlKilderkin) [reply src/Core.php:32]: That will be addressed in #34 [here](https://github.com/ucsc/ucsc-custom-functionality/pull/34/files#diff-3822a790a5cac9e3e980820c9ad0d8184339090889a97e65ea3b1a81a63ab52fR116-R118)

File: src/Blocks/Query_Loop.php (hunk 1 above)
Comment (MlKilderkin) [reply src/Blocks/Query_Loop.php]: Updated

**Pull request merged**